### PR TITLE
Extract shells

### PR DESCRIPTION
**Pull request by arnaudbore**

### CODE REVIEW

File: nextflow.config [hunk 1]
```diff
@@ -69,6 +69,12 @@ params {
         dwi_resolution=1
         dwi_interpolation="lin"
 
+    //**Extract DTI shells**//
+        maximum_dti_shell_value=1200
+
+    //**Extract fODF shells**//
+        minimum_fodf_shell_value=1000
```
Comment (GuillaumeTh) [inline nextflow.config]: Be consistent with the doc --fodf_shells.

File: USAGE [hunk 2]
```diff
@@ -92,13 +92,24 @@ OPTIONAL ARGUMENTS (current value)
...
+                                            Please write them between quotes e.g. (--fodf_shells "0 1000").
+
 --run_t1_denoising                          Run T1 denoising ($run_t1_denoising).
 
 --run_resample_t1                           Run resample T1 ($run_resample_t1).
 --t1_resolution                             T1 resolution ($t1_resolution).
 --t1_interpolation                          T1 interpolation method [nn, lin, quad, cubic] ($t1_interpolation).
 
 --number_of_tissues                         Number of tissue classes ($number_of_tissues).
+--thr_wm_seg
```
Comment (GuillaumeTh) [inline USAGE]: Please add documentation

**Review by GuillaumeTh — CHANGES_REQUESTED** [review]
I didn't review the processes because I guessed that they were not finished. You seems to don't use your extracted shells.

**Comment by arnaudbore** [comment]
Ready for testing. Should the min or max shell value be inclusive or exclusive ? @GuillaumeTh @mdesco

**Comment by mdesco** [comment]
> 

Inclusive, i.e <=

File: USAGE [hunk 3]
```diff
@@ -92,6 +87,14 @@ OPTIONAL ARGUMENTS (current value)
 --dwi_resolution                            DWI resolution ($dwi_resolution).
 --dwi_interpolation                         DWI interpolation method [nn, lin, quad, cubic] ($dwi_interpolation).
 
+--max_dti_shell_value                       Maximum shell threshold to be consider as a DTI shell ($max_dti_shell_value).
```
Comment (grahamlittlephd) [inline USAGE]: Maybe you should state here that max_dti_shell_value is used by default, and dti_shells will overide max_dti_shell_value.  Not necessary but the multiple options might confuse a user

**Comment by grahamlittlephd** [comment]
Finished testing this on openpain and Pentera 3T data.  Behaves as expected!

Nice work! I would say it's good to go!

**Comment by arnaudbore** [comment]
@GuillaumeTh do you want to give it a try before I merge ?

**Pull request merged**